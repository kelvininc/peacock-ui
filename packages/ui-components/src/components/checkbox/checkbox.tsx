import { Component, Event, EventEmitter, h, Host, Prop } from '@stencil/core';
import { EOtherIconName } from '../icon/icon.types';
import { ICheckbox, ICheckboxEvents } from './checkbox.types';

/**
 * @part icon - The icon element.
 */
@Component({
	tag: 'kv-checkbox',
	styleUrl: 'checkbox.scss',
	shadow: true
})
export class KvCheckbox implements ICheckbox, ICheckboxEvents {
	/** @inheritdoc */
	@Prop({ reflect: true }) checked?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) disabled?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) indeterminate?: boolean = false;

	/** @inheritdoc */
	@Event() clickCheckbox: EventEmitter<void>;

	private getIconName = () => {
		if (this.indeterminate) {
			return EOtherIconName.IconIndeterminateState;
		}

		if (this.checked) {
			return EOtherIconName.IconCheckState;
		}

		return EOtherIconName.IconUncheckState;
	};

	private onClick = () => {
		if (!this.disabled) {
			this.clickCheckbox.emit();
		}
	};

	render() {
		return (
			<Host
				onClick={this.onClick}
				class={{
					disabled: this.disabled
				}}
			>
				<kv-icon name={this.getIconName()} part="icon" />
			</Host>
		);
	}
}
