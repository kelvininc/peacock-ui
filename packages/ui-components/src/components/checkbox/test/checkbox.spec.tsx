import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { KvCheckbox } from '../checkbox';

describe('Radio Button (unit tests)', () => {
	let page: SpecPage;
	let component: KvCheckbox;

	describe('when rendering with default props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvCheckbox],
				html: '<kv-checkbox></kv-checkbox>'
			});
			component = page.rootInstance;
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should initialize `checked` with value false', () => {
			expect(component.checked).toBeFalsy();
		});

		it('should initialize `disabled` with value false', () => {
			expect(component.disabled).toBeFalsy();
		});

		it('should initialize `indeterminate` with empty string', () => {
			expect(component.indeterminate).toBeFalsy();
		});
	});
});
