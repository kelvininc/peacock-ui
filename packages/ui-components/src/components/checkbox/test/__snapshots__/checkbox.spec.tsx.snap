// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button (unit tests) when rendering with default props should match the snapshot 1`] = `
<kv-checkbox>
  <mock:shadow-root>
    <kv-icon name="icon-uncheck-state" part="icon"></kv-icon>
  </mock:shadow-root>
</kv-checkbox>
`;
